src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,79 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { BehaviorSubject, Observable, throwError } from 'rxjs';
+import { catchError, pluck, tap } from 'rxjs/operators';
+import { Response, User } from 'src/app/shared/models/shared.models';
+import { environment } from 'src/environments/environment';
+
+import { SessionStorageService } from './session-storage.service';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class AuthService {
+    private readonly isAuthorized$$ = new BehaviorSubject<boolean>(false);
+    public readonly isAuthorized$: Observable<boolean> =
+        this.isAuthorized$$.asObservable();
+
+    constructor(
+        private http: HttpClient,
+        private sessionStorageService: SessionStorageService,
+        private router: Router,
+    ) {
+        this.isLoggedIn();
+    }
+
+    private isLoggedIn(): void {
+        this.getToken() !== null && this.setIsAuthorized(true);
+    }
+
+    public login(user: Pick<User, 'email' | 'password'>): void {
```

Comment:
I suggest return $ from the method and subscribe to it in of login execution , lets say in app component on init - depends on implementation